playlist.c

```diff
@@ -2547,6 +2597,29 @@ playlist_t *playlist_init(const char *path, size_t size)
 
    playlist_read_file(playlist, path);
 
+   size_t i;
+   settings_t* settings = config_get_ptr();
+
+   /* try use relative paths if enabled */
+   bool use_relative_path = (settings != NULL)
+      && settings->bools.playlist_save_relative_paths
+      && !string_is_empty(settings->paths.directory_menu_content);
+   if (use_relative_path)
+   {
+      for (i = 0; i < playlist-> size; i++)
+      {
+         struct playlist_entry* entry = playlist->entries + i;
+         if (!string_is_empty(entry->relative_path))
+         {
+            if (!entry->path)
+               entry->path = (char*)malloc(PATH_MAX_LENGTH);
```

Comment:
No, we can't assign a string of `PATH_MAX_LENGTH` to every playlist entry - that's an incredible waste of memory!

`strdup()` a local char array here, instead.